### ISSUE TITLE
Add field.Other type 

### ISSUE THREAD

**Issue by cliedeman** [facebook/ent#1105]
Create `field.Other` for fields that do not fit inside standard types. E.g. postgres range or spatial types

- [x] I have searched the [issues](https://github.com/facebook/ent/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

Currently we often use String to map custom go types. E.g.

PostGis Example

Current 
```
field.String("point").
            GoType(&spatial.Point{}),
```

Desired
```
field.Other("point").
            GoType(&spatial.Point{}),
```

Postgres Range (Feels awkward)
```
field.String("daterange").
            GoType(&pgxtype.Tstzrange{}),
```

## Motivation 🔦

Other will not have any predicates
Other will error if the go type is missing
The String type will generate the normal predicates because the underlying type is expected to be a string type
E.g. `type Role string`

**Comment by a8m** [comment]
Hey @cliedeman, I agree that this option is needed, but 2 comments from my side for now - I prefer `field.Other` for the builder, because we already have a `field.TypeOther`. Also, we need to think on some elegant way to define predicates in the schema. Maybe to combine it with external-templates somehow. 

I'll think about this in the weekend, and hopefully come up with more details/thoughts from my side.

**Comment by cliedeman** [comment]
Thanks @a8m I have updtaed the issue from Custom -> Other.

I believe #1051 would also be fixed by this approach

**Comment by a8m** [comment]
Closed with #1145 and #1218. Thanks for your contribution @cliedeman 🙏
